qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -69,16 +68,14 @@ def run(self, dag):
                 continue
 
             new_circs = []
-            q = QuantumRegister(1, "q")
-            qc = QuantumCircuit(1)
-            for gate in run:
-                qc._append(gate.op, [q[0]], [])
-            operator = Operator(qc)
+            operator = Operator(run[0].op)
+            for gate in run[1:]:
+                operator = operator.compose(gate.op)
             for decomposer in self.basis:
                 new_circs.append(decomposer(operator))
             if new_circs:
                 new_circ = min(new_circs, key=lambda circ: circ.depth())
-                if qc.depth() > new_circ.depth():
+                if len(run) > new_circ.depth():
```

Comment:
```suggestion
                if len(run) > len(new_circ):
```
and similarly in the line above?

Reply:
Yeah, in both places makes sense

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5682/commits/7b6965a4c89c71557370d02e78439c8a0d37abc0